packages/engine/src/worker/mod.rs

```diff
@@ -420,7 +421,12 @@ impl WorkerController {
             };
 
             if task.pending_groups.is_empty() {
-                entry.remove();
+                let mut task = entry.remove();
+                let final_task_message = if task.final_task_messages.len() > 1 {
```

Comment:
I'd rather just always call `combine_task_messages` for consistency and ease of debugging.

Reply:
I agree about checking if it's distributed